src/rulesets/Base/arraymath.jl

```diff
@@ -28,12 +28,12 @@ function rrule(
         return (
             NoTangent(),
             InplaceableThunk(
-                @thunk(Ȳ * B'),
-                X̄ -> mul!(X̄, Ȳ, B', true, true)
+                @thunk(project(A, Ȳ * B')),
+                X̄ -> project(A, mul!(X̄, Ȳ, B', true, true))
```

Comment:
I don't think this line is right.
mul has already mutated X bar,
X bar already must have had the right structure so we are fine.

Possibly we needed to project one or both of the inputs to the `mul!`